src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -143,6 +146,12 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
                 }
 
                 _socket = connectTask.Result;
+                if (_socket == null)
```

Comment:
In the above code, if isInfiniteTimeout is false and the Wait times out, the asynchronous connection attempt will still potentially be happening in the background; is that ok?